frontend/src/components/DropboxChooser/index.tsx

```diff
@@ -0,0 +1,68 @@
+import loadable from "@loadable/component";
+import React, { FC, useState } from "react";
+
+declare global {
+  interface Window {
+    Dropbox: {
+      isBrowserSupported: () => boolean;
+      choose: (options: unknown) => void;
+    };
+  }
+}
+
+const AsyncContent = loadable(
+  () =>
+    /*webpackChunkName: 'DropboxChooser/Content' */ import(
+      "src/components/DropboxChooser/components/Content"
+    )
+);
+
+type DropboxFile = {
+  link: string;
+};
+
+const DROPBOX_OPTIONS = {
+  extensions: [".h5ad"],
+  sizeLimit: 30 * 1024 * 1024 * 1024, // 30GB
+};
+
+export interface Props {
+  onSelectUploadLink: React.Dispatch<React.SetStateAction<DropboxFile["link"]>>;
+}
+
+const DropboxChooser: FC<Props> = ({ children, onSelectUploadLink }) => {
+  const [isContentShown, setIsContentShown] = useState(false);
+
+  const handleMouseOver = () => {
+    setIsContentShown(true);
+  };
+
+  const handleClick = () => {
+    if (!window.Dropbox.isBrowserSupported()) {
+      alert("Sorry, your browser does not support Dropbox upload :(");
+    }
+
+    const options = {
+      ...DROPBOX_OPTIONS,
+      success(files: DropboxFile[] = []) {
+        // (thuang): We need to set `dl=1` to be direct download
+        const link = files[0].link.replace("dl=0", "dl=1");
```

Comment:
This is handled on the backend as well if you want to avoid repetition.

Reply:
ooh perfect thank you! I removed the edit here!